### PR TITLE
feat(security): add additional validations for a secure URL

### PR DESCRIPTION
**Pull request by maryiam**
Additional validations for a secure URL: urlContainsProtocol, omitCredentialsFromUrl and omitQueryStringFromUrl

### CODE REVIEW

File: ts/lib/url-parser.test.ts [hunk 1]
```diff
@@ -587,4 +587,83 @@ describe('UrlUtils', () => {
             urls.forEach(url => UrlUtils.isUrl(url).should.be.false());
         });
     });
+
+    describe('findUsedProtocol', () => {
+        it('should return the used url in the URL', () => {
```
Comment (Alexandre-Herve) [inline ts/lib/url-parser.test.ts]: Weird sentence!

File: ts/lib/url-parser.test.ts [hunk 2]
```diff
@@ -587,4 +587,83 @@ describe('UrlUtils', () => {
...
+
+    describe('findUsedProtocol', () => {
+        it('should return the used protocol in the URL', () => {
+            const usedProtocols = [
+                'http://',
+                'ftp://',
+                'smb://'
+            ];
+            const urls = [
+                `${usedProtocols[0]}//example.com/over/there`,
```
Comment (fellowseb) [inline ts/lib/url-parser.test.ts]: I know it's just an example but there a two unneeded slashes here.  Should we still be returning the protocol in that case ?

Comment (fellowseb) [reply ts/lib/url-parser.test.ts]: Apparently `tldts` fails to parse `http:////www.domain.com` for example.

File: package.json [hunk 3]
```diff
@@ -14,7 +14,8 @@
   "typings": "js/index",
   "scripts": {
     "build": "tsc -p ts",
-    "test": "mocha -r ts-node/register ts/**/*.test.ts",
+    "test": "TS_NODE_COMPILER_OPTIONS='{\"target\":\"es6\"}' mocha -r ts-node/register ts/**/*.test.ts",
+    "test-watch": "npm test -- --watch-extensions ts --watch",
```
Comment (fellowseb) [inline package.json]: Nice addition

File: ts/lib/url-parser.test.ts [hunk 4]
```diff
@@ -587,4 +587,83 @@ describe('UrlUtils', () => {
...
+                .should.be.true());
+        });
+
+        it('should return false if the url contains a non valid protocol', () => {
+            const urls = [
+                'example.com/page',
+                '/path/to/page',
+                'website',
+                'http:/website.com',
+                'http//website.com'
```
Comment (bradcush) [inline ts/lib/url-parser.test.ts]: Another url to test could like a valid protocol but not be one we accept (eg.  `notaccepted://`)

Comment (maryiam) [reply ts/lib/url-parser.test.ts]: done

**Pull request merged**